src/game/bosses/boss_4.c

```diff
@@ -0,0 +1,1106 @@
+            sub->tail[i].dx = Q(5.75);
+            sub->tail[i].dy = 0;
+            sub->tail[i].status = 0;
+        }
+    }
+}
+END_NONMATCH
+
+static void AeroEgg_UpdatePartsAfterBossDefeated(AeroEgg *boss)
+{
+    AeroEggSub *sub = &boss->sub;
+    s32 res;
+    u8 i;
+
+    sub->unk68 = 1;
+    sub->unk00++;
+
+    AE_UPDATE_PART_DEFEATED(sub, sub->body, 13, -70, 100,
+                            AE_DEFEATED_COLL_BODY(sub, sub->body));
+
+    for (i = 0; i < ARRAY_COUNT(sub->tail); i++) {
+        AE_UPDATE_PART_DEFEATED(sub, sub->tail[i], 7, -6, 10,
+                                AE_DEFEATED_COLL_PART(sub->tail[i]));
+    }
+
+    AE_UPDATE_PART_DEFEATED(sub, sub->tailTip, 9, -6, 10,
+                            AE_DEFEATED_COLL_PART(sub->tailTip));
+}
+
+static bool32 sub_80423EC(AeroEgg *boss)
+{
+    Sprite *s = &boss->sub.sprPilot;
+
+    bool32 result = FALSE;
+
+    if (boss->main.lives > 0) {
+        boss->main.lives--;
+
+        if (boss->main.lives & 0x1) {
+            m4aSongNumStart(SE_143);
+        } else {
+            m4aSongNumStart(SE_235);
+        }
+
+        boss->main.unk16 = 30;
+
+        if (boss->main.lives == 0) {
+            s->graphics.anim = SA2_ANIM_AERO_EGG_PILOT;
+            s->variant = 4;
+
+            INCREMENT_SCORE(1000);
+        } else {
+            s->graphics.anim = SA2_ANIM_HAMMERTANK_PILOT;
+            s->variant = 2;
+        }
+        s->prevVariant = -1;
+    } else {
+        result = TRUE;
+    }
+
+    if (gCurrentLevel != LEVEL_INDEX(ZONE_FINAL, ACT_XX_FINAL_ZONE)) {
+        if (boss->main.lives == 4) {
+            gUnknown_030054A8.unk1 = 0x11;
+        }
+    }
+
+    return result;
+}
+
+static void sub_80424EC(AeroEgg *boss)
+{
+    Sprite *s = &boss->sub.sprPilot;
+
+    if (boss->main.unk16 != 0) {
+        boss->main.unk15 = 0;
+
+        if (--boss->main.unk16 == 0) {
+            if (boss->main.lives == 0) {
+                s->graphics.anim = SA2_ANIM_AERO_EGG_PILOT;
+                s->variant = 4;
+            } else {
+                s->graphics.anim = SA2_ANIM_HAMMERTANK_PILOT;
+                s->variant = 0;
+            }
+
+            s->prevVariant = -1;
+        }
+    } else if (boss->main.unk15 != 0) {
+        boss->main.unk15--;
+
+        if (boss->main.unk15 == 0) {
+            s->graphics.anim = SA2_ANIM_HAMMERTANK_PILOT;
+            s->variant = 0;
+            s->prevVariant = -1;
+        }
+    }
+}
+
+static void sub_8042560(AeroEgg *boss)
+{
+    u8 i;
+
+    if (boss->main.unk16 != 0) {
+        for (i = 0; i < ARRAY_COUNT(gUnknown_080D7F54[PAL_BOSS_4_DEFAULT]); i++) {
+            gObjPalette[128 + i] = gUnknown_080D7F54[((gStageTime & 0x2) >> 1)][i];
+        }
+    } else {
+        for (i = 0; i < ARRAY_COUNT(gUnknown_080D7F54[PAL_BOSS_4_HIT]); i++) {
+            gObjPalette[128 + i] = gUnknown_080D7F54[PAL_BOSS_4_HIT][i];
+        }
+    }
+
+    gFlags |= FLAGS_UPDATE_SPRITE_PALETTES;
+}
+
+void AeroEggResetPos(s32 dx, s32 dy)
```

Comment:
I don't agree with the reset name? It moves the boss by dX, dY?

Reply:
It moves it back to the original spawn position, once it reaches a certain x coordinate, that's what I was trying to express with the name.